mm2src/coins/lp_coins.rs

```diff
@@ -5751,3 +5755,81 @@ pub mod for_tests {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20ApproveRequest {
+    coin: String,
+    spender: EthAddress,
+    amount: BigDecimal,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20AllowanceRequest {
+    coin: String,
+    spender: EthAddress,
+}
+
+#[derive(Debug, Deserialize, Display, EnumFromStringify, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum Erc20CallError {
+    #[display(fmt = "No such coin {}", coin)]
+    NoSuchCoin { coin: String },
+    #[display(fmt = "Coin not supported {}", coin)]
+    CoinNotSupported { coin: String },
+    #[from_stringify("NumConversError")]
+    #[display(fmt = "Invalid param: {}", _0)]
+    InvalidParam(String),
+    #[from_stringify("TransactionErr")]
+    #[display(fmt = "Transaction error {}", _0)]
+    TransactionError(String),
+    #[from_stringify("Web3RpcError")]
+    #[display(fmt = "Web3 RPC error {}", _0)]
+    Web3RpcError(String),
+}
+
+impl HttpStatusCode for Erc20CallError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            Erc20CallError::NoSuchCoin { .. }
+            | Erc20CallError::CoinNotSupported { .. }
+            | Erc20CallError::InvalidParam(_)
+            | Erc20CallError::TransactionError(_)
+            | Erc20CallError::Web3RpcError(_) => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+type Erc20AllowanceResult = MmResult<BigDecimal, Erc20CallError>;
+type Erc20ApproveResult = MmResult<BytesJson, Erc20CallError>;
+
+/// Call allowance for ERC20 tokens
+/// Returns BigDecimal value
+pub async fn allowance_rpc(ctx: MmArc, req: Erc20AllowanceRequest) -> Erc20AllowanceResult {
+    let coin = lp_coinfind_or_err(&ctx, &req.coin)
+        .await
+        .mm_err(|_| Erc20CallError::NoSuchCoin { coin: req.coin.clone() })?;
+    match coin {
+        MmCoinEnum::EthCoin(eth_coin) => {
+            let wei = eth_coin.allowance(req.spender).compat().await?;
+            let amount = u256_to_big_decimal(wei, eth_coin.decimals())?;
+            Ok(amount)
+        },
+        _ => Err(MmError::new(Erc20CallError::CoinNotSupported { coin: req.coin })),
+    }
```

Comment:
```suggestion
    match lp_coinfind_or_err(&ctx, &req.coin).await {
        Ok(MmCoinEnum::EthCoin(eth_coin)) => {
            let wei = eth_coin.allowance(req.spender).compat().await?;
            let amount = u256_to_big_decimal(wei, eth_coin.decimals())?;
            Ok(amount)
        },
        Ok(_) => Err(MmError::new(Erc20CallError::CoinNotSupported { coin: req.coin })),
        Err(_) => Err(MmError::new(Erc20CallError::NoSuchCoin { coin: req.coin.clone() })),
    }
```

Reply:
done in 5ec045f64fd1548c3c0f29fa289474ba33386561